lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2419,41 +2419,34 @@ def plot_wireframe(self, X, Y, Z, *, axlim_clip=False, **kwargs):
         tX, tY, tZ = np.transpose(X), np.transpose(Y), np.transpose(Z)
 
         if rstride:
-            rii = list(range(0, rows, rstride))
+            rii = np.arange(0, rows, rstride)
             # Add the last index only if needed
             if rows > 0 and rii[-1] != (rows - 1):
-                rii += [rows-1]
+                rii = np.append(rii, rows-1)
```

Comment:
You should be able to calculate whether the last index is included and immediately set up an array of the right size. I suspect that could be faster than `np.append`. Something like this (code untested)
```
if (rows - 1) % rstride == 0:
    # last index is hit: rri[-1] == rows -1
    rri = np.arange(0, rows, rstride)
else:
    # add the last index
    rri = np.arange(0, rows+rstride, rstride)
    rri[-1] = rows -1
```